csharp/ql/src/semmle/code/csharp/exprs/Creation.qll

```diff
@@ -296,6 +296,28 @@ class ImplicitDelegateCreation extends DelegateCreation, @implicit_delegate_crea
   override string getAPrimaryQlClass() { result = "ImplicitDelegateCreation" }
 }
 
+/**
+ * A `with` expression called on a record.
+ */
+class WithExpr extends Call, @with_expr {
```

Comment:
This should probably be a `QualifiableExpr`, where the qualifier in `x with { ... }` is `x`. In this case `x` should be at index `-1` and `{ ... }` should be at index 0.

Reply:
Should it be both a `Call` and a `QualifiableExpr`, like `MethodCall`?

The following
```
var a = new R1(1);
var b = a with { I = 2 };

record R1(int I){}
```
is equivalent to 
```
R1 r = new R1(1);
R1 r2 = r.<Clone>$();
r2.I = 2;
```

Reply:
Given that we are modelling `with` as a call to `<Clone>$` (followed by assignments), the call to the clone method should have a qualifier, which should be the LHS of the `with` expression. So yes, it should be like a `MethodCall`.